### ISSUE TITLE
Allow user to view transactions associated with a contact

### ISSUE THREAD

**Issue by AureliaKhorsand** [fecgov/fecfile-web-app#66]
Business Reason

As a Committee Administrator, I will be able to view all transaction activity for a contact as they may be needed for internal auditing purposes or calculating aggregate.

Acceptance Criteria

***Given*** I have selected to view a contact through Contact Management

 ***Then*** I will be able to see all transactions associated with that contact with the following information:

*   Transaction Type
*   Form Type _(E.g, Form 3X, Form 24)_
*   Report type
*   Date
*   Amount

***And*** if the user clicks on the transaction, it will open to that transaction in the relevant report.

### QA Notes ###
* If there are multiple dates such as in a Schedule E, this table should follow the same logic as the transactions table on the Manage transactions page. For example, for a SE transaction, if there is one date that is what is shown, if there are two, it's the Date of Disbursement.

### DEV Notes ###
- If there are multiple dates such as in a Schedule E, this table should follow the same logic as the transactions table on the Manage transactions page. For example, for a SE transaction, if there is one date that is what is shown, if there are two, it's the Date of Disbursement.

### Design ###
!image-20260113-143733.png|width=681,alt="image-20260113-143733.png"!

 See full ticket and images here: [FECFILE-1332](https://fecgov.atlassian.net/browse/FECFILE-1332)

**Comment by AureliaKhorsand** [comment]
Note: There is current functionality to select _multiple_ contacts and view all transaction activity associated with all selected contacts. That is out of scope for this user story.

**Comment by WiseQA** [comment]
Per our meeting this morning, this ticket will need put back to Product Backlog until we get our DEV branch and testing branch set up.

**Comment by AureliaKhorsand** [comment]
@WiseQA Dependencies on other tickets won't block a ticket from getting to ASA. The dependency is ticket #40 and is linked here. We just won't be able to work on it until #40 is done.

**Comment by AureliaKhorsand** [comment]
**Backlog Grooming 1/20 Notes:** Need roles and permissions. List out every transaction.

**Comment by AureliaKhorsand** [comment]
Waiting on list of OP1 transaction types from Ryan.

**Comment by mary-tcg** [comment]
@MitchellTCG @AureliaKhorsand @GreggMoreland wireframes added

**Comment by AureliaKhorsand** [comment]
@MitchellTCG @mary-tcg looks good to me from a requirements perspective. I'm assuming that if the user clicks on a transaction, it takes them to that transaction within the relevant report. I'll leave any other comments to you Mitchell, from a design perspective.

**Comment by AureliaKhorsand** [comment]
@mary-tcg can you add a column for Form Type

**Comment by AureliaKhorsand** [comment]
@mary-tcg could we see an example of a negative transaction? Also, I believe it is "48 hour" not "48 hours". Also, if we want to add sorting, confirming that will be a _separate_ ticket.

**Comment by mary-tcg** [comment]
@AureliaKhorsand @MitchellTCG wire updated

**Comment by MitchellTCG** [comment]
Updates made

**Comment by toddlees** [comment]
passes CR moving to QA

**Comment by MitchellTCG** [comment]
Placed on hold until the UX finished reviewing all table values and headings

**Comment by MitchellTCG** [comment]
Please make the following updates:
- [ ] 'Form type' column should be populated with Form Name like on the 'Manage Reports' page. ex. "Form 24"
- [ ] 'Report type' column should be populated with Name of the report like in the 'Type of report' column on the "Manage reports" page. ex. "APRIL 15 (Q1)"
- [ ] Change the format of the Date from "2016-02-03" (YEAR-MONTH-DAY) to "01/01/2016" (MONTH/DAY/YEAR)

**Comment by sasha-dresden** [comment]
For the 'Report type' column, what should it say for Form 99 and Form 1M forms? Currently on the Manage Reports page field those are empty for those. Also, I updated the Report Page so Form 24 will show either 24 HOUR or 48 HOUR to match with this.

I could just leave those blank for now and we could address it later. I created a getter in the Report class which is set for the Form3X and the Form 24, so it would just be a matter of updating the Form 99 and 1M when we get that information decided upon.

**Comment by mjtravers** [comment]
Passes CR. Sending to QA.

**Comment by WiseQA** [comment]
QA review verified when selecting a Contact s from the menu bar Mange Contacts page is displayed.

Select Contact from the contacts table.   Verified Transaction History is displayed for that contact. 

<img width="1111" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeVBVQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d8a91a746cc322b1db75412e4497666343fcd791/image.png" alt="image.png" />

_________________________________________________________________________________________________________________

Verified column headings Transaction type; Form type; Report type; Date; Amount

<img width="1351" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeVRVQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--517180dc0a5317ce58d87a848b07108bc5a87d39/image.png" alt="image.png" />

___________________________________________________________________________________________

<img width="1450" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMVBVQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d8729eddcb9c6d05eef4ff24848ba926dd1c3af1/image.png" alt="image.png" />
_____________________________________________________________________________________________

Verified when selecting a "Transaction type" from "Transaction history" table the correct transaction is displayed (tested using "Contribution to Candidate (Redesignation to)"

<img width="1322" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeWZVQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--9a4fa3743329fc33f57c7fc028a51cf49cea431b/image.png" alt="image.png" />
___________________________________________________________________________________


<img width="1424" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeXZVQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c9ed86b59e83fa07cd23697c85628dac1549580e/image.png" alt="image.png" />

____________________________________________________________________________________________________

QA Review Completed.   Moved to Stage Ready.
